src/Tizen.Multimedia.Util/ThumbnailExtractor/ThumbnailExtractor.cs

```diff
@@ -242,5 +239,117 @@ private static Native.ThumbnailExtractCallback GetCallback(TaskCompletionSource<
                 tcs.TrySetCanceled();
             });
         }
+
+        /// <summary>
+        /// Extracts the thumbnail for the given media with the specified path and size.
+        /// The generated thumbnail will be returned in <see cref="ThumbnailExtractionResult"/>.
+        /// </summary>
+        /// <remarks>
+        /// If you want to access internal storage, you should add privilege http://tizen.org/privilege/mediastorage. <br/>
+        /// If you want to access external storage, you should add privilege http://tizen.org/privilege/externalstorage.
+        /// </remarks>
+        /// <privilege>http://tizen.org/privilege/mediastorage</privilege>
+        /// <privilege>http://tizen.org/privilege/externalstorage</privilege>
+        /// <param name="path">The path of the media file to extract the thumbnail.</param>
+        /// <param name="size">The size of the thumbnail.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="path"/> is null.</exception>
+        /// <exception cref="FileNotFoundException"><paramref name="path"/> does not exist.</exception>
+        /// <exception cref="InvalidOperationException">An internal error occurs.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///     The width or the height of <paramref name="size"/> is less than or equal to zero.
+        /// </exception>
+        /// <exception cref="FileFormatException">The specified file is not supported.</exception>
+        /// <returns>The result of extracting operation.</returns>
+        /// <since_tizen> 6 </since_tizen>
+        public static ThumbnailExtractionResult Extract(string path, Size size)
```

Comment:
`ExtractAsync(string path)` is allow omission of size, but it is not, Why it is different from async?

Reply:
Native function requirement is different.
If user doesn't set size, it'll be set by default size in async set of functions. 
(thumbnail_util_set_size())
But native async functions are all deprecated and it'll deprecated in C# also.

Current sync APIs require size parameter by mandatory in native.

Reply:
It is code of native API, it allow `-1` is it bug or intended?
``` c
550	thumbnail_util_retvm_if((width > 2000 || width == 0) || (height > 2000 || width == 0), THUMBNAIL_UTIL_ERROR_INVALID_PARAMETER, "Wrong width/height");
```
https://code.sec.samsung.net/codegrok/server/tizen/preview/src/thumbnail_util.c?def=y&included_paths=&keyword=thumbnail_util_extract_to_buffer&limit=50&line=543&project=p-tizen-main&ref=20e7909918e559666883f769d5a39fd557681af2&repo=r-platform%3A%3Acore%3A%3Aapi%3A%3Athumbnail-util

Reply:
Application developer don't want care about size of image, just want to get original size. 
I hope to you provide way to not set a size

Reply:
I think it's not problem that result size should be 

> It is code of native API, it allow `-1` is it bug or intended?
> 
> ```c
> 550	thumbnail_util_retvm_if((width > 2000 || width == 0) || (height > 2000 || width == 0), THUMBNAIL_UTIL_ERROR_INVALID_PARAMETER, "Wrong width/height");
> ```
> 
> https://code.sec.samsung.net/codegrok/server/tizen/preview/src/thumbnail_util.c?def=y&included_paths=&keyword=thumbnail_util_extract_to_buffer&limit=50&line=543&project=p-tizen-main&ref=20e7909918e559666883f769d5a39fd557681af2&repo=r-platform%3A%3Acore%3A%3Aapi%3A%3Athumbnail-util

It's unsigned in native.
thumbnail_util_extract_to_buffer(const char *path, unsigned int width, unsigned int height,~~)

And Please review C# code only in here.

Reply:
> Application developer don't want care about size of image, just want to get original size.
> I hope to you provide way to not set a size

There's no API for setting default value in sync API. but it's useful as you said. 
So I'll make the method without `Size` and it'll use default value 320x240(it's original default value).